static/app/utils/profiling/hooks/useCurrentProjectFromRouteParam.tsx

```diff
@@ -1,7 +1,8 @@
+import {Project} from 'sentry/types';
 import {useParams} from 'sentry/utils/useParams';
 import useProjects from 'sentry/utils/useProjects';
 
-export function useCurrentProjectFromRouteParam() {
+export function useCurrentProjectFromRouteParam(): Project | null {
   const params = useParams();
   const projects = useProjects({limit: 1, slugs: [params?.projectId]});
   return projects.projects?.[0] ?? null;
```

Comment:
Why did this fail to infer?

Reply:
It's because we don't have `noUncheckedIndexedAccess`, so it thinks the type will ALWAYS be `Project`.